src/frontend/components/Loader/index.spec.tsx

```diff
@@ -1,14 +1,72 @@
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import React from 'react';
 
-import { Loader } from '.';
+import { Loader, Spinner } from '.';
+
+describe('<Spinner />', () => {
+  it('renders a spinner as an accessible status region', () => {
+    render(
+      <Spinner>
+        <span>Loading some object...</span>
+      </Spinner>,
+    );
+
+    const region = screen.getByRole('status', {
+      name: 'Loading some object...',
+    });
+    expect(region).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('renders a spinner as an acessible alert region', () => {
+    render(
+      <Spinner role="alert">
+        <span>Loading some important object...</span>
+      </Spinner>,
+    );
+
+    const region = screen.getByRole('alert', {
+      name: 'Loading some important object...',
+    });
+    expect(region).toHaveAttribute('aria-live', 'assertive');
+  });
+
+  it('renders a spinner that is hidden from the accessibility tree', () => {
+    render(<Spinner aria-hidden={true} />);
+    expect(screen.queryByRole('status')).toBeNull();
+    expect(screen.queryByRole('alert')).toBeNull();
+  });
+});
 
 describe('<Loader />', () => {
```

Comment:
You are testing twice the spinner, you are not testing the loader. You should only test that the loader is rendered, not every cases ?

Reply:
What would you think of this?

```
for (const Component of [Spinner, Loader]) {
  describe(Component />, () => {
    // do the tests
  })
}
```

Reply:
So there's no code duplication if this is what's bothering you

Reply:
(I updated the test to this)

Reply:
No it's just you are not testing the component itself but its children. If you modify later the component you will probably see no errors in the tests.

Reply:
But what else is there to test besides what I'm already testing?

`<Loader />` is basically some styles around `<Spinner />`. I don't see what I could test besides the accessible properties of the component

Reply:
If what you expect is styles around spinner, then test this. Because it's what you expect.

Reply:
But the existing tests did not test styling at all and there's no relevant stuff to test for styling.

What I expect:
- `<Loader />` should display an accessible spinner component for the various cases

What I test:
- `<Loader />` should display an accessible spinner component for the various cases

IMO this is true regardless of the implementation detail which is that it uses `<Spinner />` internally. Practically I would not do this if the tests took 2k lines but in this case it's simple to test both so that's what I did.

I guess the alternative I see is to just not test `<Loader />`, then if it stops using `<Spinner />` for some reason no tests will break and that's not a great outcome either

Reply:
Maybe we should get a third opinion 😅 @jbpenrath do you have something to say about this?

Reply:
From coverage point of view, you are testing both Loader and Spinner components and its fine.
But yes in fact as Loader just wraps Spinner, you are testing Spinner twice.

Reply:
> Is there something to check that <Loader /> contains a component with displayName Spinner as children ?

IMO that's testing implementation details, which I generally try to avoid. `testing-library` also explicitly does not contain any tools for this kind of test as they do not recommend them.